lib/floki/selector/parser.ex

```diff
@@ -267,4 +267,8 @@ defmodule Floki.Selector.Parser do
     Logger.debug("Only simple selectors are allowed in :not() pseudo-class. Ignoring.")
     nil
   end
+
+  defp optimize_selector(selector) do
```

Comment:
I think it worth to add a comment here that we are doing this due to an improvement in the `Selector` algo. I think the description of the test is good enough.

> Reorders classes in selector to improve matching performance.

 WDYT?

Reply:
Done e4a8a8a9c2c6106fe7c5c23543c39f2969a5b57f